### PR TITLE
Fixes to tests due to (possible) new ``tolineno`` behaviour

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

These are the changes necessary for the tests to allow https://github.com/PyCQA/astroid/pull/1351 to pass the `pylint` test suite.

I'll annotate the changes to explain why something changed and why I think this is not a bad thing.

Obviously, this is all dependent on https://github.com/PyCQA/astroid/pull/1351 actually landing, which will need to be discussed in the `astroid` repository.

_Expect a push to https://github.com/PyCQA/astroid/pull/1351 in a couple of minutes to undraft that PR._

### CODE REVIEW

File: tests/functional/ext/broad_try_clause/broad_try_clause_extension.txt
Comment (DanielNoord) [inline tests/functional/ext/broad_try_clause/broad_try_clause_extension.txt:3]: This is due to the change of no longer adding the `finally:` child to the end line number calculation of a `TryExcept` node.

File: tests/functional/i/invalid/invalid_name.py
Comment (DanielNoord) [inline tests/functional/i/invalid/invalid_name.py:67]: This actually fixes the long standing issue that is mentioned in the comment here.

File: tests/functional/r/regression/regression_2913.py
Comment (DanielNoord) [inline tests/functional/r/regression/regression_2913.py:17]: This was actually inconsistent with how we describe this should work, see:
https://github.com/PyCQA/pylint/blob/b31dd5623966069386b9df3ccc08e2834aca621d/pylint/utils/file_state.py#L70-L85

According to this, a disable on L18 should affect L17. Apparently it did not, and this got fixed with this change.

The original issue actually discusses this as well:
https://github.com/PyCQA/pylint/issues/2913

The reporter wonders why this isn't behaving like our internal comment is telling it should. I think this is thus a good change (and perhaps worthy of a changelog).

File: tests/functional/i/invalid/invalid_name.py
Comment (Pierre-Sassoulas) [reply tests/functional/i/invalid/invalid_name.py:67]: #119 is closed as 'won't fix' but apparently it was not specific to python 3.6 ?

Comment (DanielNoord) [reply tests/functional/i/invalid/invalid_name.py:67]: I think it's a combination of `parser` changes and how we do `tolineno` right now (as in, without `end_lineno`). In any case, this change clearly fixes it 😄

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas This can be rerun after the release of `astroid` and then merged. We will need to merge this first as all test suites will fail without these changes.

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you for fixing this, astroid 2.10.0 is released let's merge 😄

File: tests/checkers/unittest_stdlib.py
Comment (Pierre-Sassoulas) [inline tests/checkers/unittest_stdlib.py:53]: This actually make the formatting better, I like this. But it also means some work when upgrading 😿

**Comment by DanielNoord** [comment]
This is actually pretty problematic. Since `tolineno` is now version dependent, disables have become version dependent as well...
We might need a hotfix for `astroid` or change how we determine the position of disables. I'll investigate.

**Comment by DanielNoord** [comment]
Pushed a PR to `astroid` to fix the underlying issue. Tested against the problem here and that was resolved 😄

**Comment by z3ntu** [comment]
What's the status on this? On Alpine Linux edge pylint is currently broken because astroid 2.10 is in the repos but pylint isn't yet compatible with that. 
Is this patch ready to be picked into Alpine or is something still holding this up?

**Comment by Pierre-Sassoulas** [comment]
> What's the status on this? 

We're working on it, we're going to release astroid 2.11 and this is the first MR we'll merge once this is done.

> pylint is currently broken because astroid 2.10

You should use pylint 2.12.2 with the astroid version defined in our packaging (<2.10, but that should not matter). Don't try to set the version astroid yourself and pylint won't break.

File: tests/checkers/unittest_unicode/unittest_bad_chars.py
Comment (DanielNoord) [inline tests/checkers/unittest_unicode/unittest_bad_chars.py:133]: This is potentially a regression, but it could also be something we did not pick up previously and are now incorrectly picking up (if you understand what I mean).

The warning makes _some_ sense, but perhaps we should allow this use of ``cast``

Comment (Pierre-Sassoulas) [reply tests/checkers/unittest_unicode/unittest_bad_chars.py:133]: Yeap, I think astroid got better so a new false positive appeared in pylint (or is it ? It seems to make sense). Happens often when upgrading astroid in fact 😄

**Review by Pierre-Sassoulas — APPROVED** [review]
Happy to finally be upgrading astroid 🎉 Thank you @DanielNoord !

File: tests/checkers/unittest_unicode/unittest_bad_chars.py
Comment (DanielNoord) [reply tests/checkers/unittest_unicode/unittest_bad_chars.py:133]: Yeah, but I don't understand why it happened `2.10>2.11`. Like, we did nothing for this in `2.11`, right? Well anyway, doesn't hurt too much.

_And I don't think I can take waiting for a 2.11.1 hotfix anymore_

Comment (Pierre-Sassoulas) [reply tests/checkers/unittest_unicode/unittest_bad_chars.py:133]: >And I don't think I can take waiting for a 2.11.1 hotfix anymore

We definitely need to release soon, there are hundred of issues fixed in 2.13, some of them get reopened as duplicate frequently.

**Pull request merged**
